Mathlib/Tactic/Simps/Basic.lean

```diff
@@ -1105,11 +1106,11 @@ partial def addProjections (nm : Name) (type lhs rhs : Expr)
     fun proj ↦ !(proj.getString ++ "_").isPrefixOf x then
     let simpLemma := nm.appendAfter x
     let neededProj := (x.splitOn "_")[0]!
-    throwError "Invalid simp lemma {simpLemma}. Structure {str} does not have projection {""
-      }{neededProj}.\nThe known projections are:\n  {projs}\nYou can also see this information {""
-      }by running\n  `initialize_simps_projections? {str}`.\nNote: these projection names might {""
-      }be customly defined for `simps`, and could differ from the projection names of the {""
-      }structure."
+    throwError "Invalid simp lemma {simpLemma}. Structure {str} does not have projection \
+      {neededProj}.\nThe known projections are:\n  {projs}\nYou can also see this information \
+      by running\n  `initialize_simps_projections? {str}`.\nNote: these projection names might \
+      be customly defined for `simps`, and could differ from the projection names of the \
+      structure."
```

Comment:
```suggestion
    throwError "Invalid simp lemma {simpLemma}. \
      Structure {str} does not have projection {neededProj}.\n\
      The known projections are:\n  \
        {projs}\n\
      You can also see this information by running\n  \
        `initialize_simps_projections? {str}`.\n\
      Note: these projection names might be customly defined for `simps`, \
      and could differ from the projection names of the structure."
```

Reply:
There needs to be a `toMessageData` in there to use `indentD`, so here's another suggestion to be able to add it to the batch. I also am not sure about stowing away indentation on the previous line, so here's an alternative.

```suggestion
    throwError "Invalid simp lemma {simpLemma}. \
      Structure {str} does not have projection {neededProj}.\n\
      The known projections are:\
      {indentD <| toMessageData projs}\n\
      You can also see this information by running\
      \n  `initialize_simps_projections? {str}`.\n\
      Note: these projection names might be customly defined for `simps`, \
      and could differ from the projection names of the structure."
```